File: Q7/Q7.ts
// Calculate the nth Fibonacci number.

function FibonacciTS(n: number): number {
  if (n <= 1) return n;

  return FibonacciTS(n - 1) + FibonacciTS(n - 2);
}

console.log(FibonacciTS(10));

File: Q7/Q7.go
// Calculate the nth Fibonacci number.
package main

import (
	"fmt"
)

func fibonacci(n int) int {
	if n <= 1 {
		return n
	}

	a, b := 0, 1
	for i := 2; i <= n; i++ {
		a, b = b, a+b
	}
	return b
}

func main() {
	var n int
	fmt.Print("Enter the position (n): ")
	fmt.Scan(&n)

	result := fibonacci(n)
	fmt.Printf("Fibonacci number at position %d is: %d\n", n, result)
}
